src/BenchmarkDotNet/Validators/ConfigCompatibilityValidator.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using BenchmarkDotNet.Configs;
+
+namespace BenchmarkDotNet.Validators
+{
+    public class ConfigCompatibilityValidator : IValidator
+    {
+        public static readonly ConfigCompatibilityValidator FailOnError = new ConfigCompatibilityValidator();
+
+        public bool TreatsWarningsAsErrors => true;
+
+        public IEnumerable<ValidationError> Validate(ValidationParameters validationParameters)
+        {
+            var benchmarks = validationParameters.Benchmarks;
```

Comment:
nit: this local variable is used only once, it could be inlined to shorten the source code